docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+and expressions that can be evaluated to produce other string values.
+
+Traditionally, you needed to use positional parameters in a method
+like `string.Format`:
+
+[!code-csharp[stringFormat](../../samples/snippets/csharp/new-in-6/oldcode.cs#stringFormat)]
+
+With C# 6, the new string interpolation feature enables you to embed
+the expressions in the format string. Simple preface the string with
+`$`:
+
+[!code-csharp[stringInterpolation](../../samples/snippets/csharp/new-in-6/newcode.cs#FullNameExpressionMember)]
+
+This initial example used variable expressions for the substituted
+expressions. You can expand on this syntax to use any expression. For
+example, you could compute a student's grade point average as part of
+the interpolation:
+
+[!code-csharp[stringInterpolationExpression](../../samples/snippets/csharp/new-in-6/newcode.cs#stringInterpolationExpression)]
+
+Running the above example, you would find that the output for `Grades.Average()`
+might have more decimal places than you would like. The string interpolation
+syntax supports all the format strings available using earlier formatting
+methods. You add the format strings inside the braces. Add a `:` following
+the expression to format:
+
+[!code-csharp[stringInterpolationFormat](../../samples/snippets/csharp/new-in-6/newcode.cs#stringInterpolationFormat)]
+
+The above line of code will format the value for `Grades.Average()` as
+a floating point number with 2 decimal places.
+
+The `:` is always interpreted as the separator between the expression
+being formatted and the format string. This can introduce problems when
+your expression uses a `:` in another way, such as a conditional operator:
+
+```csharp
+public string GetGradePointPercentages() =>
+    $"Name: {LastName}, {FirstName}. G.P.A: {Grades.Any() ? Grades.Average() : double.NaN:F2}";
+```
+
+Above, the `:` is parsed as the beginning of the format string, not part
+of the conditional operator. In all cases where this happens, you can
+surround the expression with parentheses to force the compiler to interpret
+the expression as you intend:
+
+[!code-csharp[stringInterpolationConditional](../../samples/snippets/csharp/new-in-6/newcode.cs#stringInterpolationConditional)]
+
+There aren't any limitations on the expressions you can place between
+the braces. You can execute a complex LINQ query inside an interpolated
+string to perform computations and display the result:
+
+[!code-csharp[stringInterpolationLinq](../../samples/snippets/csharp/new-in-6/newcode.cs#stringInterpolationLinq)]
+
+You can see from the above sample that you can even nest a string interpolation
+expression inside another string interpolation expression. The example
+above is very likely more complex than you would want in production code.
+Rather, it is illustrative of the breadth of the feature. Any C# expression
+can be placed between the curly braces of an interpolated string.
+
+### String Interpolation and Specific Cultures
```

Comment:
let's try to make the new topics with the right section casing so it's less work to fix those in the future :-)